fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import copy
+import numpy as np
+
+from fairlearn.metrics import DemographicParity, BoundedGroupLoss
+from fairlearn.reductions.reductions_learner import ReductionsLearner
+from fairlearn.reductions.grid_search import QualityMetric, GridSearchResult
+
+
+class GridSearch(ReductionsLearner):
+    """Learner to perform a grid search given a blackbox algorithm.
+    The supplied algorithm must implement a method
+    fit(X, Y, sample_weight=[...])
+    At the present time, the only disparity metrics supported
+    are demographic parity (for classification) and bounded group
+    loss (for regression)
+    """
+
+    _KW_PROTECTED_ATTRIBUTE = "protected_attribute"
```

Comment:
I usually put these outside classes, although I can't exactly find any resource that says so. Maybe I've been doing it all wrong all this time? Just wondering if you have any opinions...

Reply:
Since it's for the kwargs for the `fit` method of this class, I thought it made more sense here. Having said that, we could well end up using at least some of these in many places.

Reply:
Okay sure, maybe they'll move to a constants file eventually then. That's great.